index.js

```diff
@@ -73,24 +63,16 @@ function getFileBlob (filePath) {
 
     chunkStore = chunkStore || new IdbChunkStore(torrentMeta.pieceLength, {name: torrentMeta.infoHash})
 
-    return downloaded.get(hash).then(isDownloaded => {
-      if (isDownloaded) {
-        let stream = ChunkStream.read(chunkStore, chunkStore.chunkLength, {
-          length: torrentMeta.length
-        })
-        return new Promise(function (resolve, reject) {
-          toBlob(stream, function (err, blob) {
-            if (err) return reject(err)
-            resolve(blob.slice(fileInfo.offset, fileInfo.offset + fileInfo.length))
-          })
-        })
-      } else {
-        // Defer until the file finishes downloading
-        return new Promise(function (resolve) {
-          if (!waitingFetches[hash]) waitingFetches[hash] = []
-          waitingFetches[hash].push(resolve)
-        })
-      }
+    priority.add(hash) // TODO only add if necessary
```

Comment:
If the file is already downloaded. It's just an unnecessary add, that immediately get's removed by the downloader. On the other hand, the downloader is the only one that knows if a file is downloaded or not so maybe the comment isn't needed anymore